src/services/utilities.ts

```diff
@@ -1717,7 +1717,19 @@ namespace ts {
 
     export function getSynthesizedDeepCloneWithRenames<T extends Node>(node: T, includeTrivia = true, renameMap?: Map<Identifier>, checker?: TypeChecker, callback?: (originalNode: Node, clone: Node) => any): T {
         let clone;
-        if (isIdentifier(node) && renameMap && checker) {
+        if (renameMap && checker && isBindingElement(node) && isIdentifier(node.name) && isObjectBindingPattern(node.parent)) {

```

Comment:
Is `parent` actually intended to be safe to use here? Is there another way to distinguish a `BindingElement` between an array binding element and an object binding element?

Reply:
This is correct

Reply:
Everytime you are using semantic operation (that is program exists the binding is done in services layer explicitly)